pallets/staking_with_credit/src/lib.rs

```diff
@@ -0,0 +1,3601 @@
+    type NextNewSession: EstimateNextNewSession<Self::BlockNumber>;
+
+    /// The number of blocks before the end of the era from which election submissions are allowed.
+    ///
+    /// Setting this to zero will disable the offchain compute and only on-chain seq-phragmen will
+    /// be used.
+    ///
+    /// This is bounded by being within the last session. Hence, setting it to a value more than the
+    /// length of a session will be pointless.
+    type ElectionLookahead: Get<Self::BlockNumber>;
+
+    /// The overarching call type.
+    type Call: Dispatchable + From<Call<Self>> + IsSubType<Call<Self>> + Clone;
+
+    /// Maximum number of balancing iterations to run in the offchain submission.
+    ///
+    /// If set to 0, balance_solution will not be executed at all.
+    type MaxIterations: Get<u32>;
+
+    /// The threshold of improvement that should be provided for a new solution to be accepted.
+    type MinSolutionScoreBump: Get<Perbill>;
+
+    /// The maximum number of nominators rewarded for each validator.
+    ///
+    /// For each validator only the `$MaxNominatorRewardedPerValidator` biggest stakers can claim
+    /// their reward. This used to limit the i/o cost for the nominator payout.
+    type MaxNominatorRewardedPerValidator: Get<u32>;
+
+    /// A configuration for base priority of unsigned transactions.
+    ///
+    /// This is exposed so that it can be tuned for particular runtime, when
+    /// multiple pallets send unsigned transactions.
+    type UnsignedPriority: Get<TransactionPriority>;
+
+    /// Weight information for extrinsics in this pallet.
+    type WeightInfo: WeightInfo;
+
+    type RewardAdjustFactor: Get<u128>;
+
+    type RewardPerBlock: Get<u128>;
+
+    type RewardAdjustPeriod: Get<u32>;
+
+    type BlocksPerEra: Get<u128>;
+
+    type RemainderMiningReward: Get<u128>;
+}
+
+/// Mode of era-forcing.
+#[derive(Copy, Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum Forcing {
+    /// Not forcing anything - just let whatever happen.
+    NotForcing,
+    /// Force a new era, then reset to `NotForcing` as soon as it is done.
+    ForceNew,
+    /// Avoid a new era indefinitely.
+    ForceNone,
+    /// Force a new era at the end of all sessions indefinitely.
+    ForceAlways,
+}
+
+impl Default for Forcing {
+    fn default() -> Self {
+        Forcing::NotForcing
+    }
+}
+
+// A value placed in storage that represents the current version of the Staking storage. This value
+// is used by the `on_runtime_upgrade` logic to determine whether we run storage migration logic.
+// This should match directly with the semantic versions of the Rust crate.
+#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, RuntimeDebug)]
+enum Releases {
+    V1_0_0Ancient,
+    V2_0_0,
+    V3_0_0,
+    V4_0_0,
+}
+
+impl Default for Releases {
+    fn default() -> Self {
+        Releases::V4_0_0
```

Comment:
should we change to other version?

Reply:
should we change to version "V1_0_0_FROM_STAKING_V4_0_0" or "V5_0_0", "V1_0_0"?

Reply:
Let's use V1_0_0. There is a line need change as well: StorageVersion build(|_: &GenesisConfig<T>| Releases::V4_0_0): Releases;  Not sure if there are more places to change.